### PR TITLE
Implemented Level select screen

### PR DESCRIPTION
**Pull request by Melanie-Gao**

### CODE REVIEW

File: core/src/edu/cornell/gdiac/temporary/GDXRoot.java [hunk 1]
```diff
@@ -171,6 +196,12 @@ public void exitScreen(Screen screen, int exitCode) {
 			setScreen(menu);
 			menu.reset();
 			menu.show();
+		}
+		// TODO: @Melanie, make this use exit code
+		else if (exitCode==ExitCode.TO_LEVEL){
+			loading = new LoadingMode("assets.json", canvas,1);
+			loading.setScreenListener(this);
+			setScreen(loading);
```
Comment (mluo24) [inline core/src/edu/cornell/gdiac/temporary/GDXRoot.java]: This is an extraneous if condition btw

File: core/src/edu/cornell/gdiac/temporary/GDXRoot.java [hunk 2]
```diff
@@ -144,15 +150,34 @@ public void exitScreen(Screen screen, int exitCode) {
...
-			playing.readLevel(directory);
+			String fileName = levelscreen.getSelectedJson();
+//			System.out.println(fileName);
+			playing.readLevel(directory,fileName);
 			playing.populate(directory);
 			playing.initializeOffset(calibration.getOffset());
 			setScreen(playing);
 			playing.show();
-		} else if (exitCode == ExitCode.TO_EDITOR) {
+		}
```
Comment (mluo24) [inline core/src/edu/cornell/gdiac/temporary/GDXRoot.java]: I'm not sure which one this is, but I'm assuming TO_LEVEL is actually to the game? I would at the very least rename them and recomment in ExitCode

**Review by mluo24 — COMMENTED** [review]
I think you'll need to modify the line in GameMode that uses listener.exitScreen to go back to the level main menu for now (which should be pause once gurvir implements it). And also have a way to "reset" the level editor.

File: core/src/edu/cornell/gdiac/temporary/GDXRoot.java (hunk 2 above)
Comment (Melanie-Gao) [reply core/src/edu/cornell/gdiac/temporary/GDXRoot.java]: yeah the two are swapped at the moment.

Comment (Melanie-Gao) [reply core/src/edu/cornell/gdiac/temporary/GDXRoot.java]: ok this is changed

**Comment by Melanie-Gao** [comment]
> I think you'll need to modify the line in GameMode that uses listener.exitScreen to go back to the level main menu for now (which should be pause once gurvir implements it). And also have a way to "reset" the level editor.

I'm still a bit confused. GameMode checked if the player hits "esc" in render (didExit), then go to menu mode. I think that's the intended behavior. I think I should change menumode, but everything in exitScreen is already updated to exit to level (I know the two are swapped but I refactored it). 

I think I can leave it like that for now, and change it after Gurvir added pause mode?

**Comment by mluo24** [comment]
> > I think you'll need to modify the line in GameMode that uses listener.exitScreen to go back to the level main menu for now (which should be pause once gurvir implements it). And also have a way to "reset" the level editor.
> 
> I'm still a bit confused. GameMode checked if the player hits "esc" in render (didExit), then go to menu mode. I think that's the intended behavior. I think I should change menumode, but everything in exitScreen is already updated to exit to level (I know the two are swapped but I refactored it).
> 
> I think I can leave it like that for now, and change it after Gurvir added pause mode?

Sorry, I looked at the wrong line and was mistaken. I'll emphasize my latter half of the advice in that you should have a way to "reset" the level editor. Right now, you have states for the buttons like I did with MenuMode, however, the problem I faced was if you did not reset those states, it was impossible to get back to the right mode because the buttons were still pressed. Take a look at my `reset` method in MenuMode to see what I mean.

**Comment by Melanie-Gao** [comment]
Thanks! It should be fixed now with the latest commit.

**Comment by mluo24** [comment]
> Thanks! It should be fixed now with the latest commit.

much better! can you also just add back marya's level somewhere so we can play it and we should be good to go.

**Comment by Melanie-Gao** [comment]
OK, I added their level as 1.json.

**Review by mluo24 — APPROVED** [review]

**Pull request merged**